File: JavaScript/0071_set_matrix_zeros/solution.js
function set_matrix_zeros(matrix) {
    let zero_indices = new Set();
    for (let arr of matrix) {
        for (let index = 0; index < arr.length; index ++) {
            if (arr[index] === 0) {
                zero_indices.add(index);
            }
        }
    }
    for (let arr of matrix) {
        if (arr.includes(0)) {
            for (let index = 0; index < arr.length; index ++) {
                arr[index] = 0;
            }
        }
    }
    for (let index of zero_indices) {
        for (let arr of matrix) {
            arr[index] = 0;
        }
    }
    return matrix;
}

const matrix = [[0,1,2,0],[3,4,5,2],[1,3,1,5]];
console.log(set_matrix_zeros(matrix));

File: Python/0071_set_matrix_zeros/solution.py
def set_matrix_zeros(matrix):
    zero_indices = set()
    for arr in matrix:
        for index in range(len(arr)):
            if arr[index] == 0:
                zero_indices.add(index)
    for arr in matrix:
        if 0 in arr:
            for index in range(len(arr)):
                arr[index] = 0
    for index in zero_indices:
        for arr in matrix:
            arr[index] = 0
    return matrix

matrix = [[0,1,2,0],[3,4,5,2],[1,3,1,5]]
print(set_matrix_zeros(matrix))